src/expr/src/explain/text.rs

```diff
@@ -298,15 +298,42 @@ impl MirRelationExpr {
                     })?;
                 }
             }
-            LetRec { ids, values, body } => {
-                let bindings = ids.iter().zip(values.iter()).collect::<Vec<_>>();
+            LetRec {
+                ids,
+                values,
+                max_iters,
+                body,
+            } => {
+                assert_eq!(ids.len(), values.len());
+                assert_eq!(ids.len(), max_iters.len());
+                let bindings = itertools::izip!(ids.iter(), values.iter(), max_iters.iter())
+                    .collect::<Vec<_>>(); // CLion needs these explicit types
                 let head = body.as_ref();
 
+                // Determine whether all `max_iters` are the same number.
+                // If all of them are the same, then we print it on top of the block (or not print
+                // it at all if it's None). If there are differences, then we print them on the
+                // ctes.
+                let all_max_iters_same = max_iters
+                    .iter()
+                    .reduce(|first, i| if i == first { first } else { &None })
+                    .unwrap_or(&None);
+
                 if ctx.config.linear_chains {
-                    writeln!(f, "{}With Mutually Recursive", ctx.indent)?;
+                    write!(f, "{}With Mutually Recursive", ctx.indent)?;
```

Comment:
This entire block is untested it seems?

Reply:
Yes, because linear chains currently doesn't work with WMR, see https://github.com/MaterializeInc/materialize/issues/19012.